src/features/posts/postsSlice.ts

```diff
@@ -0,0 +1,56 @@
+/* eslint-disable no-param-reassign */
+import { createSlice, createAsyncThunk, PayloadAction } from '@reduxjs/toolkit';
+import { Post } from '../../types/Post';
+import { getUserPosts } from '../../api/posts';
```

Comment:
Verify that the `getUserPosts` function is correctly implemented and imported from the `api/posts` file.